src/EventSubscriber/SurrogateKeyGenerator.php

```diff
@@ -61,18 +73,17 @@ public function onRespond(FilterResponseEvent $event) {
    * Used when the Surrogate-Key/X-Drupal-Cache-Tags header size otherwise
    * exceeds 16 KB.
    *
+   * @deprecated Deprecated and will be removed in future versions. Use
+   *   \Drupal::service('fastly.cache_tags.hash')->cacheTagsToHashes($cache_tags); instead.
+   *
    * @param string[] $cache_tags
    *   The cache tags in the header.
    *
    * @return string[]
    *   The hashes to use instead in the header.
    */
   public static function cacheTagsToHashes(array $cache_tags) {
-    $hashes = [];
-    foreach ($cache_tags as $cache_tag) {
-      $hashes[] = substr(md5($cache_tag), 0, 3);
-    }
-    return $hashes;
+    return \Drupal::service('fastly.cache_tags.hash')->cacheTagsToHashes($cache_tags);
```

Comment:
Using \Drupal:: inside any class is not ideal - use dependency injection here in stead?